### PR TITLE
whisper: submission of port (whisper.cpp)

### PR DESCRIPTION
**Pull request by exprez135**
#### Description

Adds the whisper.cpp project to MacPorts (a port of OpenAI's whisper in C/C++). Note: this is my first contribution to MacPorts, so forgive any mistakes or bad practices. I named the project "whisper," but maybe it needs to be whisper-cpp or something like that, though whisper is easier as a name and executable. Its variants also allow users to download the models that are necessary for using the program and store them in /opt/local/share/whisper.
Replaces #25270 

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 14.5 23F79 arm64
Xcode 15.3 15E204a

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by exprez135** [comment]
@herbygillot My lack of Github skills resulted in me using a new pull request with the changes you helped suggest. Thanks so much for your assistance, and let me know if there is anything more I can do.

File: audio/whisper/Portfile [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+notes               "
+Install this port's variants to include models, or download
+your own and use the built-in `-m` feature.
+
+Remember that whisper.cpp only functions on 16-bit WAV
+files. You can use `ffmpeg -i input.mp3 -ar 16000
+output.wav` to convert the file first.
+"
+
+checksums           whisper.cpp-1.6.2.tar.gz \
```
Comment (herbygillot) [inline audio/whisper/Portfile:26]: You can use `${version}` here:

```suggestion
checksums           whisper.cpp-${version}.tar.gz \
```

**Comment by herbygillot** [comment]
@exprez135, thank you for making these changes.

Should there be a default set of variants? Perhaps the `base` one? You can do that with `default_variants`, ie., `default_variants +base`.

File: audio/whisper/Portfile [hunk 2]
```diff
@@ -0,0 +1,120 @@
+#-*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem              1.0
+PortGroup   github      1.0
+PortGroup   makefile    1.0
```
Comment (reneeotten) [inline audio/whisper/Portfile:5]: please align the lunes above

File: audio/whisper/Portfile [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+name                whisper
+categories          audio
+github.setup        ggerganov whisper.cpp 1.6.2 v
+github.tarball_from archive
+license             MIT
```
Comment (reneeotten) [inline audio/whisper/Portfile:11]: no comma between e-mail and GitHub handle

**Review by reneeotten — CHANGES_REQUESTED** [review]

File: audio/whisper/Portfile (hunk 2 above)
Comment (exprez135) [reply audio/whisper/Portfile:5]: @reneeotten Can you explain how these lines should be aligned? Should there just be one tab between each item, or should they visually align? See two images:
Screenshot 2024-08-16 at 9 34 28 PM
(1 tab between each item)
Or
Screenshot 2024-08-16 at 9 35 02 PM
(items align visually)

Comment (exprez135) [reply audio/whisper/Portfile:5]: Oh, having reviewed some other Portfiles, I think I see how it should be.

**Comment by exprez135** [comment]
> @exprez135, thank you for making these changes.
> 
> Should there be a default set of variants? Perhaps the `base` one? You can do that with `default_variants`, ie., `default_variants +base`.

I missed this comment, my apologies. I hesitated to add a default variant, since it involves downloading a significant file size (142MB for the base; or the tiny one being 75MB). One can always separately produce or have their own model files, and the tool itself is separate from them. But if you disagree, let me know.

**Review by herbygillot — APPROVED** [review]
I think this looks good as is now, and I think your comment about these large model downloads not being such a good candidate for default variants is reasonable.

Thank you @exprez135.

**Pull request merged**